src/rewrite_intrinsics.jl

```diff
@@ -0,0 +1,68 @@
+import Sugar
+using Sugar: LazyMethod
+
+function rewrite_expr_cu(m::LazyMethod, expr)
+    was_rewritten = false
+    body = first(Sugar.replace_expr(expr) do expr
+        if isa(expr, Expr)
+            args, head = expr.args, expr.head
+            if head == :call
+                func = args[1]
+                types_array = map(args[2:end]) do x
+                    Sugar.expr_type(m, x)
+                end
+                types = (types_array...,)
+                f = Sugar.resolve_func(m, func)
+                func, _was_rewritten = rewrite_for_cudanative(f, types)
+                # function arguments have to be rewritten as well!
+                args_rewritten = false
+                fargs = map(args[2:end]) do x
+                    rewr, rewritten = rewrite_expr_cu(m, x)
+                    args_rewritten = rewritten || args_rewritten
+                    rewr
+                end
+                was_rewritten = _was_rewritten || args_rewritten || was_rewritten
+                return true, Sugar.similar_expr(expr, [func, fargs...])
+            end
+        end
+        false, expr
+    end)
+    body, was_rewritten
+end
+
+"""
+Replaces recursively Julia Base functions which are defined as intrinsics in CUDAnative
+Returns the resulting function and a bool indicating, wether the function was rewritten.
+"""
+function rewrite_for_cudanative(f, types)
+    fsym = Symbol(f)
+    # if defined in CUDAnative, but doesn't match the function passed in here
+    # The probability is very high, that this is a cudanative intrinsic
```

Comment:
A `CUDA` submodule would make code look nice, eg. `CUDA.sin`. Kinda hard to move with a deprecation in place though, unless we add a hack to `@cuda` scanning for them.